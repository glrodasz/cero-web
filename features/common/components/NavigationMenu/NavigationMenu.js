import { IconLabel } from '@glrodasz/components'
import Link from 'next/link'
import React from 'react'
import { useRouter } from 'next/router'
import useBreakpoints from '../../hooks/useBreakpoints'

const pages = [
  {
    icon: 'home',
    label: 'Home',
    href: '/home',
  },
  {
    icon: 'tasks',
    label: 'Tasks',
    href: '/planning',
  },
  {
    icon: 'reports',
    label: 'Reports',
    href: '/reports',
  },
  {
    icon: 'settings',
    label: 'Settings',
    href: '/settings',
  },
]

const NavigationMenu = () => {
  const router = useRouter()
  const { isDesktop } = useBreakpoints()

  return (
    <>
      <div className="navigation-menu">
        {pages.map(({ icon, label, href }) => (
          <Link key={label} href={href}>
            <IconLabel
              icon={icon}
              label={label}
              isActive={href === router.pathname}
              direction={isDesktop ? 'horizontal' : 'vertical'}
            />
          </Link>
        ))}
      </div>
      <style jsx>{`
        .navigation-menu {
          display: flex;
          justify-content: space-evenly;
          padding: 10px 20px;
        }

        @media (min-width: 992px) {
          .navigation-menu {
            flex-direction: column;
            align-items: flex-start;
          }
        }
      `}</style>
    </>
  )
}

export default NavigationMenu
