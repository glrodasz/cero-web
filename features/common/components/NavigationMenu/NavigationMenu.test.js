import NavigationMenu from './NavigationMenu'
import { render } from '@testing-library/react'

jest.mock('next/router', () => ({
  useRouter: () => ({
    pathname: '/home',
  }),
}))

jest.mock('../../hooks/useBreakpoints', () => () => ({
  isDesktop: true,
}))

jest.mock('@glrodasz/components', () => {
  const { dummyRender } = require('../../../../utils/tests/dummyRender')
  return {
    IconLabel: dummyRender('IconLabel'),
  }
})

describe('[ features / common / components / NavigationMenu ]', () => {
  describe('when `NavigationMenu` is mounted', () => {
    it('should render', () => {
      // Arrange
      const props = {}

      // Act
      const { asFragment } = render(<NavigationMenu {...props} />)

      // Assert
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
