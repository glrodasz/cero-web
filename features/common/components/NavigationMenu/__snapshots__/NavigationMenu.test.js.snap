// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ features / common / components / NavigationMenu ] when \`NavigationMenu\` is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="jsx-2604570837 navigation-menu"
  >
    <div
      class="jsx-2604570837 menu-item"
    >
      
[IconLabel {"icon":"home","label":"Home","isActive":true,"direction":"horizontal","size":"md"} /]
    </div>
    <div
      class="jsx-2604570837 menu-item"
    >
      
[IconLabel {"icon":"tasks","label":"Tasks","isActive":false,"direction":"horizontal","size":"md"} /]
    </div>
    <div
      class="jsx-2604570837 menu-item"
    >
      
[IconLabel {"icon":"reports","label":"Reports","isActive":false,"direction":"horizontal","size":"md"} /]
    </div>
    <div
      class="jsx-2604570837 menu-item"
    >
      
[IconLabel {"icon":"settings","label":"Settings","isActive":false,"direction":"horizontal","size":"md"} /]
    </div>
  </div>
</DocumentFragment>
`;
